src/frontend-config/frontend-config.dto.ts

```diff
@@ -0,0 +1,267 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import {
+  IsArray,
+  IsBoolean,
+  IsNumber,
+  IsOptional,
+  IsString,
+  ValidateNested,
+} from 'class-validator';
+import { ServerVersion } from '../monitoring/server-status.dto';
+
+export class AuthProviders {
+  /**
+   * Is Facebook available as a auth provider?
+   */
+  @IsBoolean()
+  facebook: boolean;
+
+  /**
+   * Is GitHub available as a auth provider?
+   */
+  @IsBoolean()
+  github: boolean;
+
+  /**
+   * Is Twitter available as a auth provider?
+   */
+  @IsBoolean()
+  twitter: boolean;
+
+  /**
+   * Is at least one GitLab server available as a auth provider?
+   */
+  @IsBoolean()
+  gitlab: boolean;
+
+  /**
+   * Is DropBox available as a auth provider?
+   */
+  @IsBoolean()
+  dropbox: boolean;
+
+  /**
+   * Is at least one LDAP server available as a auth provider?
+   */
+  @IsBoolean()
+  ldap: boolean;
+
+  /**
+   * Is Google available as a auth provider?
+   */
+  @IsBoolean()
+  google: boolean;
+
+  /**
+   * Is at least one SAML provider available as a auth provider?
+   */
+  @IsBoolean()
+  saml: boolean;
+
+  /**
+   * Is at least one OAuth2 provider available as a auth provider?
+   */
+  @IsBoolean()
+  oauth2: boolean;
+
+  /**
+   * Is internal auth available?
+   */
+  @IsBoolean()
+  internal: boolean;
+}
+
+export class BannerDto {
+  /**
+   * The text that is shown in the banner
+   * @example This is a test banner
+   */
+  @IsString()
+  text: string;
+
+  /**
+   * When the banner was last changed in ms since 01.01.1970
+   * @example 1614610648000
+   */
+  @IsString()
+  timestamp: string;
```

Comment:
This should be a `Date` and then probably called `updateTime`, like in `NoteMetadataDto`.

Reply:
done